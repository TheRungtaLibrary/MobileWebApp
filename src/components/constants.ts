export const size200 = 200;

export const size640 = 640;