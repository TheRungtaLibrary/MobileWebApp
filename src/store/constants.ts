export const Action_Types = {
    GET_VEHICLE_DATA_REQUEST: 'GET_VEHICLE_DATA_REQUEST',
    GET_VEHICLE_DATA_SUCCESS: 'GET_VEHICLE_DATA_SUCCESS'
};