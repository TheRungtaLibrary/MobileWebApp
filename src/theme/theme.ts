export const theme = {
    palette: {
        primary: {
            main: '#0052cc',
        },
        secondary: {
            main: '#edf2ff',
        },
        tertiary: {
            main: '#F8F8F8',
            light: '#ffffff',
        },
        background: {
            main: '#f6ded2',
            light: '#000',
        },
        border: {
            main: '#efefef',
            light: '#808080',
        },
        text: {
            main: '#ebedf2',
        }
    },
};