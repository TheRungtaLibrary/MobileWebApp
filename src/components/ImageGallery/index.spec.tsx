import React, { FunctionComponent } from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import { ImageGallery, Props as ImageGalleryProps } from './index';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../theme/theme';
import { size640 } from '../constants';

describe('Given <ImageGallery />', () => {
  let defaultProps: ImageGalleryProps;
  let component: HTMLElement;
  let Page: FunctionComponent<any>;

  beforeAll(() => {
    defaultProps = {
      imageData: [
        {
          "uri": "img.classistatic.de/api/v1/mo-storytest/images/HqEAAOxyIPNTcg-F",
          "set": "fe4cfedffdffffffff",
        },
        {
          "uri": "img.classistatic.de/api/v1/mo-storytest/images/GpoAAOxyZzlTcg-F",
          "set": "fe4cfedffdffffffff",
        },
      ],
      title: 'title',
      desc: 'description text',
    };
  })

  beforeEach(() => {
    Page = (): JSX.Element => (
      <ThemeProvider theme={theme}>
        <ImageGallery {...defaultProps} />
      </ThemeProvider>
    );

    const { container } = render(<Page />);

    component = container;
  })

  describe('When it initializes', () => {
    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('renders as many cards as images', () => {
      const cards = screen.getAllByTestId('image-card');
      expect(cards.length).toEqual(defaultProps.imageData.length);
    });

    describe('when a card is clicked', () => {
      it('renders image modal', () => {
        const cards = screen.getAllByTestId('image-card');
        fireEvent.click(cards[0]);

        const modal = screen.getByTestId('modal-container');

        expect(modal).toBeInTheDocument();
      });

      it('renders image of size 640 inside modal', () => {
        const cards = screen.getAllByTestId('image-card');
        fireEvent.click(cards[0]);

        const modal = screen.getByTestId('modal-container');

        expect(modal).toBeInTheDocument();

        const image = screen.getByTestId('modal-image') as HTMLImageElement;
        expect(image.src).toContain(size640.toString());
      });
    });
  })
});
