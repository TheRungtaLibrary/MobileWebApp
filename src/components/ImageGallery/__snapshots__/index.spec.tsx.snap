// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <ImageGallery /> When it initializes matches snapshot 1`] = `
<div>
  <div
    class="sc-dkPtRN kwwUQi"
  >
    <div
      class="sc-hKwDye hOPITd"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        title
      </h2>
    </div>
    <div>
      <div
        class="sc-bdvvtL bdgXaM"
        data-testid="image-card"
      >
        <figure>
          <img
            alt="image"
            src="https://img.classistatic.de/api/v1/mo-storytest/images/HqEAAOxyIPNTcg-F?rule=mo-200.jpg"
          />
          <figcaption
            class="sc-gsDKAQ dCXWnf"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
            >
              photo 
              1
            </h3>
          </figcaption>
        </figure>
      </div>
      <div
        class="sc-bdvvtL bdgXaM"
        data-testid="image-card"
      >
        <figure>
          <img
            alt="image"
            src="https://img.classistatic.de/api/v1/mo-storytest/images/GpoAAOxyZzlTcg-F?rule=mo-200.jpg"
          />
          <figcaption
            class="sc-gsDKAQ dCXWnf"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
            >
              photo 
              2
            </h3>
          </figcaption>
        </figure>
      </div>
    </div>
  </div>
</div>
`;
